src/frequenz/sdk/microgrid/connection_manager.py

```diff
@@ -156,7 +173,11 @@ async def _update_client(self, server_url: str) -> None:
 
     async def _initialize(self) -> None:
         self._microgrid = await self._client.get_microgrid_info()
-        await self._graph.refresh_from_client(self._client)
+        components, connections = await asyncio.gather(
+            self._client.list_components(),
+            self._client.list_connections(),
+        )
+        self._graph = ComponentGraph(set(components), set(connections))
```

Comment:
If you want to gether even further :P
```suggestion
        self._microgrid, components, connections = await asyncio.gather(
            self._client.get_microgrid_info()
            self._client.list_components(),
            self._client.list_connections(),
        )
        self._graph = ComponentGraph(set(components), set(connections))
```

Reply:
Added an issue: https://github.com/frequenz-floss/frequenz-sdk-python/issues/1328